src/Block/Service/AbstractMenuBlockService.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\BlockBundle\Block\Service;
+
+use Knp\Menu\ItemInterface;
+use Sonata\BlockBundle\Block\BlockContextInterface;
+use Sonata\BlockBundle\Form\Mapper\FormMapper;
+use Sonata\BlockBundle\Model\BlockInterface;
+use Sonata\Form\Type\ImmutableArrayType;
+use Sonata\Form\Validator\ErrorElement;
+use Symfony\Component\Form\Extension\Core\Type\CheckboxType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormTypeInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Twig\Environment;
+
+abstract class AbstractMenuBlockService extends AbstractBlockService implements EditableBlockService
```

Comment:
At the end you choose to not add the metadata method here then? will it be on the breadcrumb classes then?

Reply:
This class is new, so I'm able to not add the metadata method here and let user implement it.

In the seo PR https://github.com/sonata-project/SonataSeoBundle/pull/692 I provided a default implementation to avoid the BC break, and we'll remove the default implementation in next major.

I still wonder if it wouldn't a good idea to split editableBlock but that can be another topic :p